waiter/integration/waiter/basic_test.clj

```diff
@@ -292,6 +317,44 @@
             (is (seq active-instances))
             (is (every? :healthy? active-instances))))))))
 
+(defn- run-backend-proto-service-test
+  "Helper method to run tests with various backend protocols"
+  [waiter-url backend-proto backend-scheme backend-proto-version]
+  (let [nginx-command (nginx-server-command backend-proto)
+        request-headers {:x-waiter-backend-proto backend-proto
+                         :x-waiter-cmd nginx-command
+                         :x-waiter-health-check-port-index 1
```

Comment:
Should we have a test that doesn't require health checks on a separate port?

Reply:
Added `test-h2-backend-proto-service-health-check-on-port0`